app/main.py

```diff
@@ -1,3 +1,68 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> None:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | (int, float)) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(
+                self.km + other.km
+            )
+        else:
+            return Distance(
+                self.km + other
+            )
+
+    def __iadd__(self, other: Distance | (int, float)) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            self.km += other
+        return self
+
+    def __mul__(self, other: (int, float)) -> Distance:
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: (int, float)) -> Distance:
+        if isinstance(other, (float, int)):
+            return Distance(round(self.km / other, 2)
+
+                            )
+
+    def __lt__(self, other: Distance | (int, float)) -> bool:
+        if isinstance(other, Distance):
+            return self.km < other.km
+        else:
+            return self.km < other
+
+    def __gt__(self, other: Distance | (int, float)) -> bool:
+        if isinstance(other, Distance):
+            return self.km > other.km
+        else:
+            return self.km > other
+
+    def __eq__(self, other: Distance | (int, float)) -> bool:
+        if isinstance(other, Distance):
+            return self.km == other.km
+        else:
+            return self.km == other
+
+    def __le__(self, other: Distance | (int, float)) -> bool:
+        if isinstance(other, Distance):
+            return self.km <= other.km
+        else:
+            return self.km <= other
```

Comment:
```suggestion
    def __le__(self, other: Distance | int | float) -> bool:
        if isinstance(other, Distance):
            return self.km <= other.km
        return self.km <= other
```
please fix everywhere

Reply:
Okay